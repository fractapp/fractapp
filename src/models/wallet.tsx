import { ImageSourcePropType } from "react-native";

export enum Currency {
    Polkadot = 0,
    Kusama
}

export class Wallet {
    name: string;
    address: string;
    balance: number;
    currency: Currency;

    symbol: string;
    usdValue: number;
    logo: ImageSourcePropType;

    constructor(name: string, address: string, currency: Currency, balance: number) {
        this.name = name;
        this.address = address;
        this.balance = balance;
        this.currency = currency;
        
        //TODO get state
        let rate = 10
        this.usdValue = balance * rate
        switch (currency) {
            case Currency.Polkadot:
                this.logo = require("assets/img/dot.png")
                this.symbol = "DOT"
                break
            case Currency.Kusama:
                this.logo = require("assets/img/kusama.png")
                this.symbol = "KSM"
                break
        }

    }
}
