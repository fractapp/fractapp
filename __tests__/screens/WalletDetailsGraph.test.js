import React, {useContext, useState} from 'react';
import renderer from 'react-test-renderer';
import {WalletDetailsGraph} from 'screens/WalletDetailsGraph';
import {Currency, Wallet} from 'types/wallet';
import GlobalStore, {initialState} from 'storage/Global';
import ChatsStore from 'storage/Chats';
import {TxStatus, TxType} from 'types/transaction';
import {fireEvent, render} from '@testing-library/react-native';

jest.mock('storage/DB', () => ({}));
jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useState: jest.fn(),
  useContext: jest.fn(),
  Linking: jest.fn(() => ({
    openURL: jest.fn(),
  })),
}));
jest.mock('utils/backend', () => ({
  getImgUrl: jest.fn(() => 'userAvatarMock'),
}));
jest.mock('react-native-i18n', () => ({
  t: (value) => value,
}));

useState.mockImplementation((init) => [init, jest.fn()]);

it('Test view with empty state', () => {
  useContext.mockReturnValueOnce({
    state: GlobalStore.initialState(),
  });
  useContext.mockReturnValueOnce({
    state: ChatsStore.initialState(),
  });

  const wallets = [
    new Wallet(
      'Wallet Polkadot',
      'address#1',
      Currency.DOT,
      100,
      '1000000000000',
      10,
      0,
    ),
    new Wallet(
      'Wallet KSM',
      'address#2',
      Currency.KSM,
      200,
      '3000000000000',
      20,
      0,
    ),
  ];

  const tree = renderer
    .create(
      <WalletDetailsGraph
        navigation={null}
        route={{
          params: {
            wallets: wallets,
          },
        }}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Test view with 1 wallet', () => {
  useContext.mockReturnValueOnce({
    state: GlobalStore.initialState(),
  });
  useContext.mockReturnValueOnce({
    state: ChatsStore.initialState(),
  });

  const wallets = [
    new Wallet(
      'Wallet Polkadot',
      'address#1',
      Currency.DOT,
      100,
      '1000000000000',
      10,
      0,
    ),
  ];

  const txs = new Map();
  txs.set(Currency.DOT, {
    id: '1',
    userId: null,
    address: 'address#1',
    currency: Currency.DOT,
    txType: TxType.Sent,
    timestamp: new Date('02-12-2020').getTime(),
    value: 12,
    usdValue: 12,
    fee: 12,
    usdFee: 12,
    status: TxStatus.Fail,
  });

  const chatsState = ChatsStore.initialState();
  chatsState.transactions.set(txs.keys(), {
    transactionById: txs.values(),
  });
  useContext.mockReturnValueOnce({
    state: {
      transactions: chatsState.transactions,
      isInitialized: true,
    },
    dispatch: () => null,
  });

  const tree = renderer
    .create(
      <WalletDetailsGraph
        navigation={null}
        route={{
          params: {
            wallets: wallets,
          },
        }}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Test view with txs', () => {
  const wallets = [
    new Wallet(
      'Wallet Polkadot',
      'address#1',
      Currency.DOT,
      100,
      '1000000000000',
      10,
      0,
    ),
    new Wallet(
      'Wallet KSM',
      'address#2',
      Currency.KSM,
      200,
      '3000000000000',
      20,
      0,
    ),
  ];

  const globalState = GlobalStore.initialState();
  globalState.users.set('userId', {
    id: 'id',
    name: 'name',
    username: 'username',
    avatarExt: 'png',
    lastUpdate: new Date('12-12-2020').getTime(),
    addresses: {
      0: 'address#2DOT',
      1: 'address#2KSM',
    },
  });

  const chatsState = ChatsStore.initialState();
  chatsState.transactions.set(Currency.DOT, {
    transactionById: new Map([
      [
        '1',
        {
          id: '1',
          userId: null,
          address: 'address#1',
          currency: Currency.DOT,
          txType: TxType.Sent,
          timestamp: new Date('02-12-2020').getTime(),
          value: 12,
          usdValue: 12,
          fee: 12,
          usdFee: 12,
          status: TxStatus.Fail,
        },
      ],
    ]),
  });
  chatsState.transactions.set(Currency.KSM, {
    transactionById: new Map([
      [
        '2',
        {
          id: '2',
          userId: 'userId',
          address: 'address#2DOT',
          currency: Currency.KSM,
          txType: TxType.Sent,
          timestamp: new Date('03-12-2020').getTime(),
          value: 10,
          usdValue: 10,
          fee: 10,
          usdFee: 10,
          status: TxStatus.Success,
        },
      ],
    ]),
  });

  useContext.mockReturnValueOnce({
    state: globalState,
    dispatch: () => null,
  });
  useContext.mockReturnValueOnce({
    state: {
      transactions: chatsState.transactions,
      isInitialized: true,
    },
    dispatch: () => null,
  });

  const tree = renderer
    .create(
      <WalletDetailsGraph
        navigation={null}
        route={{
          params: {
            wallets: wallets,
          },
        }}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Test view click txs', () => {
  const wallets = [
    new Wallet(
      'Wallet Polkadot',
      'address#1',
      Currency.DOT,
      100,
      '1000000000000',
      10,
      1,
    ),
    new Wallet(
      'Wallet KSM',
      'address#2',
      Currency.KSM,
      200,
      '3000000000000',
      20,
      2,
    ),
  ];
  const txs = new Map();
  txs.set(Currency.DOT, {
    id: '1',
    userId: null,
    address: 'address#1',
    currency: Currency.DOT,
    txType: TxType.Sent,
    timestamp: new Date('02-12-2020').getTime(),
    value: 12,
    usdValue: 12,
    fee: 12,
    usdFee: 12,
    status: TxStatus.Fail,
  });
  txs.set(Currency.KSM, {
    id: '2',
    userId: 'userId',
    address: 'address#2KSM',
    currency: Currency.KSM,
    txType: TxType.Sent,
    timestamp: new Date('03-12-2020').getTime(),
    value: 10,
    usdValue: 10,
    fee: 10,
    usdFee: 10,
    status: TxStatus.Success,
  });


  const chatsState = ChatsStore.initialState();
  chatsState.transactions.set(Currency.DOT, {
    transactionById: new Map([
      [
        '1',
        txs.get(Currency.DOT),
      ],
    ]),
  });
  chatsState.transactions.set(Currency.KSM, {
    transactionById: new Map([
      [
        '2',
        txs.get(Currency.KSM),
      ],
    ]),
  });

  const globalState = GlobalStore.initialState();
  globalState.users.set('userId', {
    id: 'id',
    name: 'name',
    username: 'username',
    avatarExt: 'png',
    lastUpdate: new Date('12-12-2020').getTime(),
    addresses: {
      0: 'address#2DOT',
      1: 'address#2KSM',
    },
  });

  useContext.mockReturnValueOnce({
    state: globalState,
    dispatch: () => null,
  });
  useContext.mockReturnValueOnce({
    state: {
      transactions: chatsState.transactions,
      isInitialized: true,
    },
    dispatch: () => null,
  });

  const navigate = jest.fn();
  const component = render(
    <WalletDetailsGraph
      navigation={{
        navigate: navigate,
      }}
      route={{
        params: {
          wallets: wallets,
        },
      }}
    />,
  );
  fireEvent.press(component.getByText('address#1'));
  expect(navigate).toBeCalledWith('TransactionDetails', {
    transaction: txs.get(Currency.DOT),
    wallet: wallets[txs.get(Currency.DOT).currency],
    user: null,
  });

  fireEvent.press(component.getByText('name'));
  expect(navigate).toBeCalledWith('TransactionDetails', {
    transaction: txs.get(Currency.KSM),
    wallet: wallets[txs.get(Currency.KSM).currency],
    user: globalState.users.get('userId'),
  });
});

