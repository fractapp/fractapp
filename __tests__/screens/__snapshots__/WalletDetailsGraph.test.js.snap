// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test view with 1 wallet 1`] = `
<RCTScrollView
  ListHeaderComponent={[Function]}
  data={Array []}
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  updateCellsBatchingPeriod={50}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
    >
      <View
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "paddingTop": 30,
              "position": "absolute",
              "width": "30%",
            }
          }
        >
          <View
            className="VictoryContainer"
            onMoveShouldSetResponder={[Function]}
            onMoveShouldSetResponderCapture={[Function]}
            onResponderEnd={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderReject={[Function]}
            onResponderRelease={[Function]}
            onResponderStart={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onStartShouldSetResponderCapture={[Function]}
            pointerEvents="box-none"
            style={
              Object {
                "position": "relative",
              }
            }
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={300}
              bbWidth={750}
              focusable={false}
              height={300}
              meetOrSlice={0}
              minX={0}
              minY={0}
              role="img"
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "height": "100%",
                    "width": "100%",
                  },
                  Object {
                    "flex": 0,
                    "height": 300,
                    "width": 750,
                  },
                ]
              }
              vbHeight={300}
              vbWidth={750}
              width={750}
            >
              <RNSVGGroup>
                <RNSVGRect
                  fill={null}
                  height={300}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  width={750}
                  x={0}
                  y={0}
                />
                <RNSVGRect
                  fill={null}
                  height={0}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  vectorEffect={1}
                  width={0}
                  x={0}
                  y={0}
                />
              </RNSVGGroup>
            </RNSVGSvgView>
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "zIndex": 99,
                }
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={300}
                bbWidth={750}
                focusable={false}
                height={300}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Object {
                      "height": "100%",
                      "overflow": "visible",
                      "width": "100%",
                    },
                    Object {
                      "flex": 0,
                      "height": 300,
                      "width": 750,
                    },
                  ]
                }
                vbHeight={300}
                vbWidth={750}
                width={750}
              >
                <RNSVGGroup />
              </RNSVGSvgView>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            className="VictoryContainer"
            onMoveShouldSetResponder={[Function]}
            onMoveShouldSetResponderCapture={[Function]}
            onResponderEnd={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderReject={[Function]}
            onResponderRelease={[Function]}
            onResponderStart={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onStartShouldSetResponderCapture={[Function]}
            pointerEvents="box-none"
            style={
              Object {
                "position": "relative",
              }
            }
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={230}
              bbWidth={750}
              focusable={false}
              height={230}
              meetOrSlice={0}
              minX={0}
              minY={0}
              role="img"
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "height": "100%",
                    "width": "100%",
                  },
                  Object {
                    "flex": 0,
                    "height": 230,
                    "width": 750,
                  },
                ]
              }
              vbHeight={230}
              vbWidth={750}
              width={750}
            >
              <RNSVGGroup>
                <RNSVGRect
                  fill={null}
                  height={230}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  width={750}
                  x={0}
                  y={0}
                />
                <RNSVGGroup />
              </RNSVGGroup>
            </RNSVGSvgView>
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "zIndex": 99,
                }
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={230}
                bbWidth={750}
                focusable={false}
                height={230}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Object {
                      "height": "100%",
                      "overflow": "visible",
                      "width": "100%",
                    },
                    Object {
                      "flex": 0,
                      "height": 230,
                      "width": 750,
                    },
                  ]
                }
                vbHeight={230}
                vbWidth={750}
                width={750}
              >
                <RNSVGGroup />
              </RNSVGSvgView>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#cccccc",
              "height": 1,
              "marginTop": 15,
              "width": "90%",
            }
          }
        />
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`Test view with empty state 1`] = `
<RCTScrollView
  ListHeaderComponent={[Function]}
  data={Array []}
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  updateCellsBatchingPeriod={50}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
    >
      <View
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "paddingTop": 30,
              "position": "absolute",
              "width": "30%",
            }
          }
        >
          <View
            className="VictoryContainer"
            onMoveShouldSetResponder={[Function]}
            onMoveShouldSetResponderCapture={[Function]}
            onResponderEnd={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderReject={[Function]}
            onResponderRelease={[Function]}
            onResponderStart={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onStartShouldSetResponderCapture={[Function]}
            pointerEvents="box-none"
            style={
              Object {
                "position": "relative",
              }
            }
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={300}
              bbWidth={750}
              focusable={false}
              height={300}
              meetOrSlice={0}
              minX={0}
              minY={0}
              role="img"
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "height": "100%",
                    "width": "100%",
                  },
                  Object {
                    "flex": 0,
                    "height": 300,
                    "width": 750,
                  },
                ]
              }
              vbHeight={300}
              vbWidth={750}
              width={750}
            >
              <RNSVGGroup>
                <RNSVGRect
                  fill={null}
                  height={300}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  width={750}
                  x={0}
                  y={0}
                />
                <RNSVGRect
                  fill={null}
                  height={0}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  vectorEffect={1}
                  width={0}
                  x={0}
                  y={0}
                />
              </RNSVGGroup>
            </RNSVGSvgView>
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "zIndex": 99,
                }
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={300}
                bbWidth={750}
                focusable={false}
                height={300}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Object {
                      "height": "100%",
                      "overflow": "visible",
                      "width": "100%",
                    },
                    Object {
                      "flex": 0,
                      "height": 300,
                      "width": 750,
                    },
                  ]
                }
                vbHeight={300}
                vbWidth={750}
                width={750}
              >
                <RNSVGGroup />
              </RNSVGSvgView>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            className="VictoryContainer"
            onMoveShouldSetResponder={[Function]}
            onMoveShouldSetResponderCapture={[Function]}
            onResponderEnd={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderReject={[Function]}
            onResponderRelease={[Function]}
            onResponderStart={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onStartShouldSetResponderCapture={[Function]}
            pointerEvents="box-none"
            style={
              Object {
                "position": "relative",
              }
            }
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={230}
              bbWidth={750}
              focusable={false}
              height={230}
              meetOrSlice={0}
              minX={0}
              minY={0}
              role="img"
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "height": "100%",
                    "width": "100%",
                  },
                  Object {
                    "flex": 0,
                    "height": 230,
                    "width": 750,
                  },
                ]
              }
              vbHeight={230}
              vbWidth={750}
              width={750}
            >
              <RNSVGGroup>
                <RNSVGRect
                  fill={null}
                  height={230}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  width={750}
                  x={0}
                  y={0}
                />
                <RNSVGGroup />
              </RNSVGGroup>
            </RNSVGSvgView>
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "zIndex": 99,
                }
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={230}
                bbWidth={750}
                focusable={false}
                height={230}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Object {
                      "height": "100%",
                      "overflow": "visible",
                      "width": "100%",
                    },
                    Object {
                      "flex": 0,
                      "height": 230,
                      "width": 750,
                    },
                  ]
                }
                vbHeight={230}
                vbWidth={750}
                width={750}
              >
                <RNSVGGroup />
              </RNSVGSvgView>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#cccccc",
              "height": 1,
              "marginTop": 15,
              "width": "90%",
            }
          }
        />
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`Test view with txs 1`] = `
<RCTScrollView
  ListHeaderComponent={[Function]}
  data={
    Array [
      Object {
        "data": Array [
          Object {
            "address": "address#2DOT",
            "currency": 1,
            "fee": 10,
            "id": "2",
            "status": 1,
            "timestamp": 1583960400000,
            "txType": 1,
            "usdFee": 10,
            "usdValue": 10,
            "userId": "userId",
            "value": 10,
          },
        ],
        "title": "12 months.march 2020",
      },
      Object {
        "data": Array [
          Object {
            "address": "address#1",
            "currency": 0,
            "fee": 12,
            "id": "1",
            "status": 2,
            "timestamp": 1581454800000,
            "txType": 1,
            "usdFee": 12,
            "usdValue": 12,
            "userId": null,
            "value": 12,
          },
        ],
        "title": "12 months.february 2020",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={
    Array [
      1,
      4,
    ]
  }
  updateCellsBatchingPeriod={50}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
    >
      <View
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "paddingTop": 30,
              "position": "absolute",
              "width": "30%",
            }
          }
        >
          <View
            className="VictoryContainer"
            onMoveShouldSetResponder={[Function]}
            onMoveShouldSetResponderCapture={[Function]}
            onResponderEnd={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderReject={[Function]}
            onResponderRelease={[Function]}
            onResponderStart={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onStartShouldSetResponderCapture={[Function]}
            pointerEvents="box-none"
            style={
              Object {
                "position": "relative",
              }
            }
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={300}
              bbWidth={750}
              focusable={false}
              height={300}
              meetOrSlice={0}
              minX={0}
              minY={0}
              role="img"
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "height": "100%",
                    "width": "100%",
                  },
                  Object {
                    "flex": 0,
                    "height": 300,
                    "width": 750,
                  },
                ]
              }
              vbHeight={300}
              vbWidth={750}
              width={750}
            >
              <RNSVGGroup>
                <RNSVGRect
                  fill={null}
                  height={300}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  width={750}
                  x={0}
                  y={0}
                />
                <RNSVGRect
                  fill={null}
                  height={0}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  vectorEffect={1}
                  width={0}
                  x={0}
                  y={0}
                />
              </RNSVGGroup>
            </RNSVGSvgView>
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "zIndex": 99,
                }
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={300}
                bbWidth={750}
                focusable={false}
                height={300}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Object {
                      "height": "100%",
                      "overflow": "visible",
                      "width": "100%",
                    },
                    Object {
                      "flex": 0,
                      "height": 300,
                      "width": 750,
                    },
                  ]
                }
                vbHeight={300}
                vbWidth={750}
                width={750}
              >
                <RNSVGGroup />
              </RNSVGSvgView>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            className="VictoryContainer"
            onMoveShouldSetResponder={[Function]}
            onMoveShouldSetResponderCapture={[Function]}
            onResponderEnd={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderReject={[Function]}
            onResponderRelease={[Function]}
            onResponderStart={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onStartShouldSetResponderCapture={[Function]}
            pointerEvents="box-none"
            style={
              Object {
                "position": "relative",
              }
            }
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={230}
              bbWidth={750}
              focusable={false}
              height={230}
              meetOrSlice={0}
              minX={0}
              minY={0}
              role="img"
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "height": "100%",
                    "width": "100%",
                  },
                  Object {
                    "flex": 0,
                    "height": 230,
                    "width": 750,
                  },
                ]
              }
              vbHeight={230}
              vbWidth={750}
              width={750}
            >
              <RNSVGGroup>
                <RNSVGRect
                  fill={null}
                  height={230}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  width={750}
                  x={0}
                  y={0}
                />
                <RNSVGGroup />
              </RNSVGGroup>
            </RNSVGSvgView>
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "zIndex": 99,
                }
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={230}
                bbWidth={750}
                focusable={false}
                height={230}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Object {
                      "height": "100%",
                      "overflow": "visible",
                      "width": "100%",
                    },
                    Object {
                      "flex": 0,
                      "height": 230,
                      "width": 750,
                    },
                  ]
                }
                vbHeight={230}
                vbWidth={750}
                width={750}
              >
                <RNSVGGroup />
              </RNSVGSvgView>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#cccccc",
              "height": 1,
              "marginTop": 15,
              "width": "90%",
            }
          }
        />
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "marginTop": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "fontFamily": "Roboto-Bold",
              "fontSize": 17,
              "fontStyle": "normal",
              "fontWeight": "normal",
              "marginBottom": 5,
              "marginLeft": "5%",
            }
          }
        >
          12 months.march 2020
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "paddingBottom": 10,
            "paddingTop": 10,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "width": "90%",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <Image
              height={50}
              source={
                Object {
                  "uri": "userAvatarMock",
                }
              }
              style={
                Object {
                  "borderRadius": 25,
                  "height": 50,
                  "width": 50,
                }
              }
              width={50}
            />
            <View
              style={
                Object {
                  "alignSelf": "center",
                  "flex": 1,
                  "flexDirection": "column",
                  "marginLeft": 10,
                }
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Object {
                    "color": "black",
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 16,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  }
                }
              >
                name
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "Roboto-Regular",
                      "fontSize": 15,
                      "fontStyle": "normal",
                    },
                    Object {
                      "color": "#EA4335",
                      "textAlign": "left",
                    },
                  ]
                }
              >
                -
                10
                 
                KSM
              </Text>
            </View>
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 15,
                    "fontStyle": "normal",
                  },
                  Object {
                    "alignSelf": "center",
                    "color": "#EA4335",
                  },
                ]
              }
            >
              -
              $
              10
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    />
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "marginTop": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "fontFamily": "Roboto-Bold",
              "fontSize": 17,
              "fontStyle": "normal",
              "fontWeight": "normal",
              "marginBottom": 5,
              "marginLeft": "5%",
            }
          }
        >
          12 months.february 2020
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "paddingBottom": 10,
            "paddingTop": 10,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "width": "90%",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../assets/img/dot.png",
                }
              }
              style={
                Object {
                  "borderColor": "black",
                  "borderRadius": 25,
                  "borderWidth": 0.8,
                  "height": 50,
                  "overflow": "hidden",
                  "width": 50,
                }
              }
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderRadius": 23,
                    "bottom": 0,
                    "height": 23,
                    "justifyContent": "center",
                    "left": 30,
                    "position": "absolute",
                    "width": 23,
                  },
                  Object {
                    "borderColor": "#EA4335",
                    "borderWidth": 1,
                  },
                ]
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#EA4335",
                      "fontSize": 20,
                    },
                    undefined,
                    Object {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                󰅖
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignSelf": "center",
                  "flex": 1,
                  "flexDirection": "column",
                  "marginLeft": 10,
                }
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Object {
                    "color": "black",
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 16,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  }
                }
              >
                address#1
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "Roboto-Regular",
                      "fontSize": 15,
                      "fontStyle": "normal",
                    },
                    Object {
                      "color": "#EA4335",
                      "textAlign": "left",
                    },
                  ]
                }
              >
                -
                12
                 
                DOT
              </Text>
            </View>
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 15,
                    "fontStyle": "normal",
                  },
                  Object {
                    "alignSelf": "center",
                    "color": "#EA4335",
                  },
                ]
              }
            >
              -
              $
              12
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    />
  </View>
</RCTScrollView>
`;
